specs/sds_detection_modules.adoc

```diff
@@ -49,47 +49,104 @@ plugins and auto detect the storage systems (if any) on underlying storage nodes
 
 * Tag the dectected details (if any) with the storage nodes details
 
-* Enhance the import cluster flows in node agent component to decide on what
-Tendrl integration module gets deployed on the underlying storage nodes
-
 === Alternatives
 
 None
 
 === Data model impact:
 
-None
+* The entity model for `Node` would look as below post tagging
+
+```
+class Node(EtcdObj):
+    __name__ = 'nodes/%s/Node'
+
+    node_id = fields.StrField("node_id")
+    fqdn = fields.StrField("fqdn")
+    status = fields.StrField("status")
+    cluster_id = fields.StrField("cluster_id")
+    cluster_role = fields.StrField("cluster_role")
+    # The cluster_attrs would contain a dictionary with keys specific to cluster
+    # e.g.
+    # - cluster_name
+    # - fsid
+    # - .....
+    cluster_attrs = fields.DictField("cluster_attrs", {'str': 'str'})
+
+    def render(self):
+        self.__name__ = self.__name__ % self.node_id
+        return super(Node, self).render()
+```
+
+* The definitions file would get additional details for detected SDS details as
+below
+
+```
+data = """---
+namespace.tendrl.node_agent:
+  objects:
+    Cpu:
+    ..................
+    Node:
+      atoms:
+        cmd:
+          enabled: true
+          inputs:
+            mandatory:
+              - Node.cmd_str
+          name: "Execute CMD on Node"
+          help: "Executes a command"
+          run: tendrl.node_agent.atoms.node.cmd.Cmd
+          type: Create
+          uuid: dc8fff3a-34d9-4786-9282-55eff6abb6c3
+        check_node_up:
+          enabled: true
+          inputs:
+            mandatory:
+              - Node.fqdn
+          outputs:
+            - Node.status
+          name: "check whether the node is up"
+          help: "Checks if a node is up"
+          run: tendrl.node_agent.atoms.node.check_node_up
+          type: Create
+          uuid: eda0b13a-7362-48d5-b5ca-4b6d6533a5ab
+      attrs:
+        cmd_str:
+          type: String
+        fqdn:
+          type: String
+        status:
+          type: Boolean
+        cluster_id:
```

Comment:
I am ok with this but my understanding was that these details need to be tagged with node inventory.
@brainfunked @r0h4n @nnDarshan comments?